### DISCUSSION TITLE
Not a correct API Key

### DISCUSSION THREAD

**Discussion by Iwhite67** [PhlexPlexico/G5V#141] [Q&A]
Hello

when i cancel match, i have this error

```L 05/22/2022 - 19:54:44: [G5WS.smx] [ERR] API request failed, HTTP status code: 500
L 05/22/2022 - 19:54:44: [G5WS.smx] [ERR] Response:
{
    "message": "Not a correct API Key."
}```

**Comment by PhlexPlexico** [comment]
are you running the game server on a windows machine? I know there was some issues awhile back with Rest in PAWN not being able to send JSON response bodies.

You can adjust the API to use a `console.log(req.body);` as well in that call (and restart the API) to see what is being sent in the body/request from the game server. If it's blank, it's either a `cURL` issue, or an issue with your network. Last time someone had this issue, they used the Docker images provided and no longer ran into the problem.

Responses can be seen in this issue - https://github.com/PhlexPlexico/G5WS/issues/37

**Reply by PhlexPlexico** [reply]
One thing I would also recommend, would be to try hitting the API directly then instead of through the reverse proxy, and see if you are still running into the issue :)

**Reply by Iwhite67** [reply]
I'm on debian

L 05/22/2022 - 20:05:56: [get5] cleanedUrl (SteamWorks) = http://match.cs.white-gaming.fr/api//matches/13/config
Match config loading initialized.
L 05/22/2022 - 20:05:56: [G5WS] get5_web_api_url now set to 
L 05/22/2022 - 20:05:56: [get5] Calling Get5_OnPreLoadMatchConfig(config=remote_config1.json)
[Zilat] Loaded match config.
L 05/22/2022 - 20:05:56: [get5] Get5_OnGameStateChanged(from=0, to=1)
L 05/22/2022 - 20:05:56: [G5WS] get5_web_api_url now set to http://match.cs.white-gaming.fr/api/
L 05/22/2022 - 20:05:56: [get5] get5_event: {
    "matchid": "13",
    "params": {
        "team1_name": "Iwhite",
        "team2_name": "Sqal"
    },
    "event": "series_start"
}
L 05/22/2022 - 20:05:56: [get5] Calling Get5_OnEvent(event name = series_start)
L 05/22/2022 - 20:05:56: [get5] Calling Get5_OnSeriesInit
L 05/22/2022 - 20:05:56: [G5WS.smx] Failed to create logo directory: resource/flash/econ/tournaments/teams
sv_load_forced_client_names_file: loaded 2 client name(s) to enforce!
Can't use cheat cvar sv_grenade_trajectory in multiplayer, unless the server has sv_cheats set to 1.
Server waking up from hibernation
sv_load_forced_client_names_file: loaded 2 client name(s) to enforce!

**Reply by PhlexPlexico** [reply]
That looks like it created the match just fine? It's complaining about the logo directories in which you just have to create them yourself since the game server may have some permission issues. Other than that, I see no problem with the creation of that match.

**Reply by PhlexPlexico** [reply]
One thing you can also check is to see if the API key is set on the game server as well.

**Reply by Iwhite67** [reply]
Logo is normal, i don't have folder but when a cancel match, i have a error with api key

**Reply by Iwhite67** [reply]
and i have this error from i have put RIP

**Reply by PhlexPlexico** [reply]
Okay, so in the API logs you'll be able to see what fails, but you also may need to add a `console.log(req.body)` call in the `/cancel` route in `matchserver.js`, right before checking the API key to see what is being sent in. 

Or, if the match is being cancelled from within the game server, I would also check the `get5_web_api_key` cvar, too.

**Reply by Iwhite67** [reply]
i cancel from Vue after create match from vue

so i will check where is the file i need to edit

**Reply by PhlexPlexico** [reply]
It will be in G5API, not G5V. As I said, in `matchserver.js` :)

**Reply by Iwhite67** [reply]
yes i know, is just for edit a file into docker i search on web actualy

**Reply by PhlexPlexico** [reply]
Okay, but also what version of the plugin are you using? I was thinking about this and if you're using an outdated API plugin, it's going to try calling cancelling the match, and then fires an event `OnSeriesEnd`. The latest release of the plugin (including the latest nightly) fixes this. If you're trying to just cancel the match, this is most likely why. So please make sure your game plugin is up to date :)

**Reply by PhlexPlexico** [reply]
https://github.com/PhlexPlexico/G5WS/releases/tag/Nightly-bee129

**Reply by Iwhite67** [reply]
> https://github.com/PhlexPlexico/G5WS/releases/tag/Nightly-bee129

Same

so what is the problem if i don't have this api key ?

**Reply by PhlexPlexico** [reply]
As long as the match gets cancelled (the game server will say that the match is cancelled by an admin), then nothing. I'm not sure as to why you'd be getting this error though, as the latest nightly + master branch of the API shouldn't be making that call anymore. What version of get5 are you using?

**Reply by Iwhite67** [reply]
![image](https://user-images.githubusercontent.com/9698964/169715092-5991511b-87a9-4922-ab6e-72ac7b261bc4.png)

**Reply by PhlexPlexico** [reply]
https://ci.splewis.net/job/get5/lastSuccessfulBuild/artifact/builds/get5/get5-545.zip

I would try the latest build, .7.2 may have some issues maybe.

**Reply by Iwhite67** [reply]
(0.8.0-dev+545)

same

**Reply by PhlexPlexico** [reply]
Can you do do `get5_debug 32` and start a match, then cancel it, and send me the game server logs?

**Reply by Iwhite67** [reply]
```get5_debug 32
hostname: Zilat: Sqal vs Iwhite
version : 1.38.3.1/13831 1459/8581 secure  [G:1:2653357] 
udp/ip  : 0.0.0.0:27015  (public ip: 82.66.157.93)
os      :  Linux
type    :  community dedicated
map     : de_dust2
gotv[0]:  port 27025, delay 105.0s, rate 128.0
players : 0 humans, 1 bots (10/0 max) (not hibernating)
# userid name uniqueid connected ping loss state rate adr
# 2 "GOTV" BOT active 128
#end
{
 "gamestate": 0,
 "available": 1,
 "plugin_version": "0.8.0-dev+545"
}
L 05/22/2022 - 20:51:28: [get5] cleanedUrl (SteamWorks) = http://match.cs.white-gaming.fr/api//matches/20/config
Match config loading initialized.
L 05/22/2022 - 20:51:28: [G5WS] get5_web_api_url now set to 
L 05/22/2022 - 20:51:28: [get5] Calling Get5_OnPreLoadMatchConfig(config=remote_config1.json)
[Zilat] Loaded match config.
L 05/22/2022 - 20:51:28: [get5] Get5_OnGameStateChanged(from=0, to=1)
L 05/22/2022 - 20:51:28: [G5WS] get5_web_api_url now set to http://match.cs.white-gaming.fr/api/
L 05/22/2022 - 20:51:28: [get5] get5_event: {
    "matchid": "20",
    "params": {
        "team1_name": "Sqal",
        "team2_name": "Iwhite"
    },
    "event": "series_start"
}
L 05/22/2022 - 20:51:28: [get5] Calling Get5_OnEvent(event name = series_start)
L 05/22/2022 - 20:51:28: [get5] Calling Get5_OnSeriesInit
L 05/22/2022 - 20:51:28: [G5WS.smx] Failed to create logo directory: resource/flash/econ/tournaments/teams
sv_load_forced_client_names_file: loaded 2 client name(s) to enforce!
Can't use cheat cvar sv_grenade_trajectory in multiplayer, unless the server has sv_cheats set to 1.
sv_load_forced_client_names_file: loaded 2 client name(s) to enforce!
sv_load_forced_client_names_file: loaded 2 client name(s) to enforce!
L 05/22/2022 - 20:51:33: [get5] Calling Get5_OnMapResult(map=de_dust2, winner=3, team1score=0, team2score=0, mapnum=-1)
{"avatar_url":"https:\/\/i.imgur.com\/rHYHg0j.png","username":"Match Result Bot","embeds":[{"fields":[],"author":{"name":"Match on de_dust2"},"footer":{"icon_url":"https:\/\/i.imgur.com\/yCw3yxZ.jpg","text":"Bot provided by Ferex"},"title":"[NOT READY] Sqal 0-0 [NOT READY] Iwhite","color":39423,"thumbnail":{"url":"https:\/\/i.imgur.com\/qCYQGjv.jpg"}}]}
L 05/22/2022 - 20:51:33: [get5discord.smx] Sending request to https://discord.com/api/webhooks/965155432102526996/vZQRFoZFMxVGw3wVn8zF6HtymIOXLqc38B9o3YKVTdfhG9M6XyBkSywlPSl-w2R0jJAy
L 05/22/2022 - 20:51:33: [G5WS] Our URL is: http://match.cs.white-gaming.fr/api/match/%d/map/%d/finish
L 05/22/2022 - 20:51:33: [G5WS] Trying to create request to url http://match.cs.white-gaming.fr/api/match/20/map/-1/finish
L 05/22/2022 - 20:51:33: [get5] get5_event: {
    "matchid": "20",
    "params": {
        "team2_series_score": 0,
        "team1_series_score": 0
    },
    "event": "series_cancel"
}
L 05/22/2022 - 20:51:33: [get5] Calling Get5_OnEvent(event name = series_cancel)
L 05/22/2022 - 20:51:33: [get5] Calling Get5_OnSeriesResult(winner=3, team1_series_score=0, team2_series_score=0)
L 05/22/2022 - 20:51:33: [G5WS] Our URL is: http://match.cs.white-gaming.fr/api/match/%d/finish
L 05/22/2022 - 20:51:33: [G5WS] Trying to create request to url http://match.cs.white-gaming.fr/api/match/20/finish
L 05/22/2022 - 20:51:33: [G5WS] get5_web_api_url now set to http://match.cs.white-gaming.fr/api/
L 05/22/2022 - 20:51:33: [get5] Get5_OnGameStateChanged(from=1, to=0)
[Zilat] An admin force ended the match.
L 05/22/2022 - 20:51:33: [G5WS] get5_web_api_url now set to 
L 05/22/2022 - 20:51:33: [get5] Calling Get5_OnDemoFinished(file=)
L 05/22/2022 - 20:51:33: [G5WS.smx] [ERR] API request failed, HTTP status code: 500
L 05/22/2022 - 20:51:33: [G5WS.smx] [ERR] Response:
{
    "message": "Not a correct API Key."
}
````

**Reply by PhlexPlexico** [reply]
Oh! It's the on demo finished call. Okay, thanks for this. This error can most definitely be ignored, but I'll make sure to push a patch out for the API. Thank you! :)

**Reply by Iwhite67** [reply]
"get5_web_api_key" = "X3UHGVGIVVIUXQ3HWAM78BM0"

![image](https://user-images.githubusercontent.com/9698964/169715365-a159f827-07e0-4fdb-a132-75d314dcce5d.png)

**Reply by Iwhite67** [reply]
ok, is a false error, my api work fine ?

**Reply by PhlexPlexico** [reply]
It should. If you'd like, could you add `console.log(req.body);` to here
https://github.com/PhlexPlexico/G5API/blob/6ac6faa0368657938a4432a0efd025d7a8c7da0e/routes/legacy/api.js#L927

To do this you can call `docker exec -it G5API bash` and use `vim` or `nano` to edit the file, save, and `yarn pm2 restart G5API` in the container. But yeah, I do believe it can be ignored :)

**Reply by Iwhite67** [reply]
G5API    | 0|G5API  | Not a correct API Key.
G5API    | 0|G5API  | POST /match/22/finish 500 17.264 ms - 36
G5API    | 0|G5API  | ::ffff:172.18.0.1 - - [22/May/2022:21:01:24 +0000] "POST /match/22/finish HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"


is see nothing in log

**Reply by PhlexPlexico** [reply]
Hmmm, are you forfeiting a match or just cancelling it?

**Reply by PhlexPlexico** [reply]
https://github.com/PhlexPlexico/G5API/blob/6ac6faa0368657938a4432a0efd025d7a8c7da0e/routes/legacy/api.js#L175

Try putting the `console.log(req.body);` here then, too :)

**Reply by Iwhite67** [reply]
is same if a cancel or forfeit.

when i have time i check to make a full match with veto

**Reply by Iwhite67** [reply]
![image](https://user-images.githubusercontent.com/9698964/169715833-d3409b25-6f03-4f20-801e-1f9c00e5adec.png)

**Reply by PhlexPlexico** [reply]
Forfeit will hit a different route than cancel. When you cancel the match, it technically should hit `/finish` but *should* return a 200 code, so I'm not sure why the API key is returning "incorrect".

Forfeiting will do something completely different, and shouldn't have the same issue

**Reply by PhlexPlexico** [reply]
Okay, are you using NGINX or something for the reverse proxy? Can you link directly to the API (no `/api` but change it to the port you use for the API) and try doing that?

**Reply by Iwhite67** [reply]
Forfeit or cancel, is all time error 500 on finish route

![image](https://user-images.githubusercontent.com/9698964/169715917-2738844d-2573-4f07-9568-b299b70e53f5.png)


all time, array are empty

**Reply by Iwhite67** [reply]
i use nginx

**Reply by PhlexPlexico** [reply]
Yeah so this was an issue before as well, with the NGINX server seemingly stripping the body. I personally use Caddy, which hasn't run into this issue. Play a few rounds and see if anything else posts the same error? It should update stats but I'm not too sure.

**Reply by Iwhite67** [reply]
http://match.cs.white-gaming.fr/match/9

my last test when i have this error 

![image](https://user-images.githubusercontent.com/9698964/169716044-4a79e033-d5ad-4fa8-9436-4bece6a342b4.png)

**Reply by Iwhite67** [reply]
when i have time this week, i will try to make a test with veto.

**Reply by PhlexPlexico** [reply]
🤷‍♂️ yeah I'm not sure, I think this is a server config issue at least but I have no idea how I would go about debugging it. If that's the only time it's calling and failing I would ignore it, personally. If everything else works as intended I would say it's fine. But very odd, nonetheless.

**Reply by Bunker56** [reply]
> when i have time this week, i will try to make a test with veto.

I tried, the same thing... POST 500 Not a correct API Key.

**Reply by PhlexPlexico** [reply]
If you wouldn't mind, could you try doing the same on the public panel? I can fetch the logs if you provide the match id. Thanks!

**Reply by Bunker56** [reply]
> check the `get5_web_api_key` cvar

In get5.cfg? I tried, the same result... And in what format? In quotes or not?

**Reply by Iwhite67** [reply]
When the match are start, put just get5_web_api_key into the server console, he return the API key

**Reply by Bunker56** [reply]
for example...
get5_web_api_key
"get5_web_api_key" = "1FF3253ED32A63DA9B025FE066FA0678" ( def. "" ) - Match API key, this is automatically set through rcon
thats it?

**Reply by PhlexPlexico** [reply]
Yup, so the API key is being set, it's just your server that is receiving a null value on the body most likely, as the API response is showing a blank body. 

I would try removing the NGINX reverse proxy and see if you can contact the API directly through the port and see if that works. I believe you would need to change the API URL to use the port specifically, as well.

**Reply by Bunker56** [reply]
I received this message in console using your Caddy configuration entirely, NGINX was completely disabled.

get5_debug 32
[Get5] Team #2 вычеркнули de_ancient
L 05/27/2022 - 22:44:15: [get5] get5_event: {
    "matchid": "29",
    "params": {
        "team": "team2",
        "map_name": "de_ancient"
    },
    "event": "map_veto"
}
L 05/27/2022 - 22:44:15: get5_event: {
    "matchid": "29",
    "params": {
        "team": "team2",
        "map_name": "de_ancient"
    },
    "event": "map_veto"
}
L 05/27/2022 - 22:44:15: [get5] Calling Get5_OnEvent(event name = map_veto)
L 05/27/2022 - 22:44:15: [get5] Calling Get5_OnMapVetoed(team=1, map=de_ancient)
L 05/27/2022 - 22:44:15: [G5WS] Map Veto START team  map vetoed de_ancient
L 05/27/2022 - 22:44:15: [G5WS] Our URL is: http://b56-match.ydns.eu/api/match/%d/vetoUpdate
L 05/27/2022 - 22:44:15: [G5WS] Trying to create request to url http://b56-match.ydns.eu/api/match/29/vetoUpdate
L 05/27/2022 - 22:44:15: [G5WS] Accepted Map Veto for team team2.
L 05/27/2022 - 22:44:15: [G5WS.smx] [ERR] API request failed, HTTP status code: 500
L 05/27/2022 - 22:44:15: [G5WS.smx] [ERR] Response:
{
    "message": "Not a correct API Key."
}

**Reply by PhlexPlexico** [reply]
As I said, please don't use a reverse proxy and try just connecting directly to the API (default port 3301). See if there is an issue with proxy management. However, I would also make sure that there are no errors in your logs as well. I've been running separate instances with no issues right now either. Are you running from the docker images, or a local pull? Is your config set correctly? There are a lot of outliers here that I'm thinking it may be pointing to a user configuration issue. In the G5API folder you should be able to run `yarn pm2 logs G5API` to see if there are any other errors coming out from the application.

**Reply by Bunker56** [reply]
I use Docker

**Reply by Iwhite67** [reply]
> I use Docker

Same, Docker for me

**Reply by PhlexPlexico** [reply]
Then do either `docker container logs G5API` to view the latest logs or 

```bash
docker container exec -it G5API bash
yarn pm2 logs G5API
```

And check the logs as you do the vetoes. Would you mind sending me your configs (with sensitive information _removed_) and I can further examine?

Thanks!

**Reply by Bunker56** [reply]
Which configs exactly?

**Reply by PhlexPlexico** [reply]
The docker-compose file, if you're using docker.

**Reply by Bunker56** [reply]
Sure...

**Comment by Iwhite67** [comment]
Yes us automatically


_________________________

Alexis Zaun
Technicien SAV Nord Est

FOM Industrie FRANCE

***@***.******@***.***>
(+33) 06 40 62 82 45
________________________________

**Comment by Iwhite67** [comment]
new test with new infra, new serv, new g5api/vue with docker

when i make a veto, i have error 404 into vetoupdate, mapstats and vetoes

![image](https://user-images.githubusercontent.com/9698964/170870799-4128e6d4-39ea-4b6f-8391-033cad25262a.png)

**Reply by Iwhite67** [reply]
no stats and score are comming to API

i have RIP 1.3.1 extension
Get5 (0.8.0-dev) by splewis
G5WS - Get5 Web Stats" (3.0.1) by phlexplexico

get5_apistats and mysqlstats are disable

**Reply by Iwhite67** [reply]
other info, me and other people see, none of the routes that are in legacy/api work, there is no console.log feedback

**Reply by Iwhite67** [reply]
Problem found, Post request inbount to nginx pass on GET Request to API

Now go work to found a solution

**Reply by Iwhite67** [reply]
SSL is the problem

now i come to same problem of other day

![image](https://user-images.githubusercontent.com/9698964/170873480-ee08423b-f242-4192-96b3-3f1747520b85.png)

and not a valid key

**Reply by PhlexPlexico** [reply]
If you can remove the sensitive information from your config file and post it here that would be great. This is honestly still sounding like a network problem.

I forget if you have done this already but can you set the API URL to point to the API directly (`http(s)://get5.yourdomain.com:3301`) and try doing that.

If you're receiving empty bodies that means there has to be something up with your ports, game server, or firewall that is preventing it from working, as I've tried on two different hosts (nginx proxy within nginx proxy, and a caddy reverse proxy, both with https) and have it working as well. So I'm not sure what exactly is causing this as I cannot find a way to reproduce.

Thanks!

**Reply by Iwhite67** [reply]
all test are made today

With Nginx and Caddy, i have empty body
when i put a API URL to 3301, i have empty body


```version: '2.1'

networks:
  get5:

services:
  caddy:
    image: caddy:2-alpine
    container_name: caddy-reverse-proxy
    restart: unless-stopped
    ports:
      - "80:80"
    volumes: 
      - ./Caddyfile:/etc/caddy/Caddyfile
    networks:
      - get5
    environment:
      - CADDY_URL=http://get5.afti-lan.fr/
      - CADDY_API_ENDPOINT=api
      - CADDY_REVERSE_PROXY_PORT=3301

  get5db:
    image: yobasystems/alpine-mariadb
    container_name: get5db
    restart: always
    environment:
      - MYSQL_ROOT_PASSWORD=
      - MYSQL_DATABASE=get5
      - MYSQL_USER=get5
      - MYSQL_PASSWORD=
    networks:
      - get5

  g5api:
    image: ghcr.io/phlexplexico/g5api:latest
    depends_on:
      - get5db
    container_name: G5API
    volumes: 
      - redisVol:/RedisFiles
    expose:
      - 3301
    environment:
      - PORT=3301
      - HOSTNAME=http://get5.afti-lan.fr/
      - DBKEY=
      - STEAMAPIKEY=
      - SHAREDSECRET=
      - CLIENTHOME=http://get5.afti-lan.fr/
      - APIURL=http://get5.afti-lan.fr/api
      - SQLUSER=get5
      - SQLPASSWORD=
      - SQLPORT=3306
      - DATABASE=get5
      - SQLHOST=get5db
      - ADMINS=
      - SUPERADMINS=
      - REDISPASSWORD=
      - UPLOADDEMOS=true
    restart: unless-stopped
    networks:
      - get5

  g5v:
    image: ghcr.io/phlexplexico/g5v:latest
    depends_on: 
      - g5api
      - caddy
    container_name: G5V-Front-End
    restart: always
    networks:
      - get5


volumes:
  redisVol:
    external: false```

**Reply by PhlexPlexico** [reply]
Change `expose: 3301` to `ports: "3301:3301` and APIURL to `http://get5.afti-lan.fr:3301`. Restart and test by going into the browser with that url to see if it loads then setup a match.

Are you using a VPS service like Linode or Digital ocean? Could you try doing the setup on one of those as well and see if it works. That way we can nail down and see if it's a game server issue, or API issue.

**Reply by Iwhite67** [reply]
is a vmware with debian 11

i test after eat with port

**Reply by Iwhite67** [reply]
![image](https://user-images.githubusercontent.com/9698964/170880209-58652f16-e7bf-4126-a86a-419d30f44f96.png)
i make my own post request, is working, so in on side of server

**Reply by Iwhite67** [reply]
can you confirm just G5WS GET5 and RIP is enough ?

![image](https://user-images.githubusercontent.com/9698964/170881425-9e6dee8c-82c4-4f5a-b7ea-10d8b2bf7f5a.png)
With postman, 5/5
With G5WS, 0/5, body keep empty

**Reply by PhlexPlexico** [reply]
What os is your fave server running?

**Comment by Iwhite67** [comment]
Debian 11

Envoyé à partir de Courrier<https://go.microsoft.com/fwlink/?LinkId=550986> pour Windows

De : ***@***.***>
Envoyé le :dimanche 29 mai 2022 18:53
À : ***@***.***>
Cc : ***@***.***>; ***@***.***>
Objet :Re: [PhlexPlexico/G5V] Not a correct API Key (Discussion #141)


What os is your fave server running?

**Reply by PhlexPlexico** [reply]
Okay, there may be a curl issue. Could you try using a different os for the game server if possible?

**Reply by Iwhite67** [reply]
> 
same with ubuntu 20.04

csgo serv are into a docker container with debian (i use pterodactyl)

i have a plugin (get5 discord integration) who make a post to discord webhooks and he working

**Reply by Iwhite67** [reply]
this is not a problem between G5WS and RIP ?

**Reply by PhlexPlexico** [reply]
That's what I'm trying to figure out. If it's sending an empty body that seems like cURLn ot working as intended on your machines. Considering that I have had this working on several servers (windows, Linux), I don't see how it could be anything related. I would like to see if you could try a different service for testing such as DatHost (game server) with your API, and then a different API (can even use the public panel) with your game servers, and see which one is failing.

**Reply by Iwhite67** [reply]
my api are working fine because i see a body when i use postman.

i thinks is on server side with docker

**Reply by Iwhite67** [reply]
test in progress without docker

**Reply by Iwhite67** [reply]
result of test with LGSM directly on Debian 11
Nothing

if a make a curl post from serv, i have body, if csgo server make curl post, nothing

**Reply by PhlexPlexico** [reply]
But can you try on something that isn't debian 11? Like a temporary VPS? I'm trying to corner the case that it may be an OS issue on the game server.

Thanks!

**Reply by Iwhite67** [reply]
> But can you try on something that isn't debian 11? Like a temporary VPS? I'm trying to corner the case that it may be an OS issue on the game server.
> 
> Thanks!


Ubuntu 20.04 without docker wip

**Reply by PhlexPlexico** [reply]
Is this on a different server as well? Like, one that isn't on the same network as the Debian machine?

**Reply by Iwhite67** [reply]
all test are made with two different network and two different machine

**Reply by Iwhite67** [reply]
Ubuntu 20.04 same, curl post work but not on serv

**Reply by PhlexPlexico** [reply]
And are you using a docker image for the CSGO same server as well? Or just LGSM? I'm not sure I can exactly help you since everything is working on my end. Could you try using the public web panel with one of your game servers and see if that works? I can get the logs as well.

**Reply by Iwhite67** [reply]
LGSM on Debian 11 Not work
LGSM on Ubuntu 20.04 Not work

All fresh install with G5WS GET5 RIP SW

**Reply by Iwhite67** [reply]
test in progress with public panel

**Reply by Iwhite67** [reply]
is working on your panel but with HTTPS

**Reply by PhlexPlexico** [reply]
That's most likely the reason why. I would attempt using HTTPS first, I never have really supported HTTP in the first place, as I assume this may be an sm-ripext bug as well using HTTP2 for calls on non HTTPS servers.

**Reply by Iwhite67** [reply]
OKAY all is working with CADDY HTTPS

but Nginx reverse proxy not working because when SSL are enable, all POST Request are change to GET

**Reply by PhlexPlexico** [reply]
Okay, I've created a ticket on the sm-ripext repository, it looks like there may be an issue when trying to perform POST requests against an HTTP service, glad to see we got the bottom of this at least, and hopefully we can find a fix!

**Reply by PhlexPlexico** [reply]
@Iwhite67 @Bunker56 would you mind trying this build with your HTTP server and see if it works?
https://github.com/ErikMinekus/sm-ripext/suites/5700970486/artifacts/187824424

Thanks!

**Reply by Bunker56** [reply]
It`s working fine! Thanks for help!..

**Reply by Bunker56** [reply]
I apologize for the immodest question, but I still didn't understand where the demos are loaded...

**Reply by PhlexPlexico** [reply]
Glad to hear it's working!!

As for demos, I believe if you have GoTV enabled, they will be stored at the root directory in the game server. The G5WS plugin has `get5_upload_demos` as a Cvar that isn't recorded. This is because if people use public panels I wouldn't want them uploading constantly. If you add `get5_upload_demos 1` to your `knife.cfg` or `live.cfg` for get5, it should upload to the API and will be available under `public/demos` in the API folder :)

Cheers!
